packages/editor/src/utils/utils.test.ts

```diff
@@ -0,0 +1,29 @@
+import { generateId, splitMacroExpressions, splitNewLine } from './utils';
+
+test('generateId', () => {
+  expect(generateId()).toEqual('0');
+  expect(generateId()).toEqual('1');
+  expect(generateId()).toEqual('2');
+});
+
+test('splitNewLine', () => {
+  expect(splitNewLine('')).toEqual(['']);
+  expect(splitNewLine('abc')).toEqual(['abc']);
+  expect(splitNewLine('abc\ndef')).toEqual(['abc', 'def']);
+  expect(splitNewLine('abc\rdef')).toEqual(['abc', 'def']);
+  expect(splitNewLine('abc\r\ndef')).toEqual(['abc', 'def']);
+});
+
+test('splitMacroExpressions', () => {
+  expect(splitMacroExpressions('')).toEqual([]);
+  expect(splitMacroExpressions('abc')).toEqual(['abc']);
+  expect(splitMacroExpressions('abc <%= ivy.cms.co("hi") %>')).toEqual(['abc ', '<%= ivy.cms.co("hi") %>']);
+  expect(splitMacroExpressions('abc <%= ivy.cms.co("hi") %>, hi')).toEqual(['abc ', '<%= ivy.cms.co("hi") %>', ', hi']);
+  expect(splitMacroExpressions('abc <%= ivy.cms.co("hi") %>, hi <%=in.accept%>')).toEqual([
+    'abc ',
+    '<%= ivy.cms.co("hi") %>',
+    ', hi ',
+    '<%=in.accept%>'
+  ]);
+  expect(splitMacroExpressions('<%= ivy.cms.co("hi") %><%=in.accept%>')).toEqual(['<%= ivy.cms.co("hi") %>', '<%=in.accept%>']);
```

Comment:
devil test '<%= ivy.cms.co("hi %>")%>'

Reply:
Does not need to be fixed for the first version. ;-)

Reply:
I'm not sure if it is really a use case that we ever support this case in this UI (even if it's technical totally fine), but I think it is not worth to fix this until someone reports this as a bug.
Unless you have the regex for it right away? 😄 
But you're right a test for this does definitely not harm 👍

Reply:
You cannot solve such a thing with a simple Regex because it needs a state (inside string or not). Have a look at ch.ivyteam.ivy.scripting.expander.impl.TextTagExpander. But as said it is not a prio one.